tests/main/interfaces-system-files/task.yaml

```diff
@@ -0,0 +1,94 @@
+summary: Ensure that the system-files interface works.
+
+details: |
+    The system-files interface allows access specific system files or directories.
+
+environment:
+    # keep in sync with tests/lib/snaps/test-snapd-sh/meta/snap.yaml
+    TESTDIR: /mnt/testdir
+
+prepare: |
+    # shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB/snaps.sh"
+    install_local test-snapd-sh
+
+    # shellcheck source=tests/lib/files.sh
+    . "$TESTSLIB/files.sh"
+
+    # Fist layer of dirs and files
+    ensure_dir_exists_backup_real "$TESTDIR"
+    ensure_file_exists_backup_real "$TESTDIR"/.testfile1
+    ensure_file_exists_backup_real "$TESTDIR"/testfile1
+    ensure_dir_exists_backup_real "$TESTDIR"/.testdir1
+    ensure_dir_exists_backup_real "$TESTDIR"/testdir1
+
+    # Second layer of dirs and files
+    ensure_file_exists_backup_real "$TESTDIR"/.testdir1/.testfilé2
+    ensure_file_exists_backup_real "$TESTDIR/.testdir1/test file2"
+    ensure_dir_exists_backup_real "$TESTDIR"/root
+
+    # Not accessible dirs and files
+    ensure_dir_exists_backup_real /root/.testdir1
+    ensure_file_exists_backup_real /root/.testfile1
+
+restore: |
+    rm -f call.error
+
+    # shellcheck source=tests/lib/files.sh
+    . "$TESTSLIB/files.sh"
+
+    clean_dir "$TESTDIR"
+    clean_dir /root/.testdira1
+    clean_dir /root/.testfile1
+
+execute: |
+    echo "The interface is not connected by default"
+    snap interfaces -i system-files | MATCH "\\- +test-snapd-sh:system-files"
+
+    echo "When the interface is connected"
+    snap connect test-snapd-sh:system-files
+
+    echo "And not other interfaces are connected"
+    snap disconnect test-snapd-sh:home
+
+    echo "Then the snap is able to access all the files and dirs in /testdir"
+    test-snapd-sh.with-system-files-plug -c "cat $TESTDIR/.testfile1" | MATCH "content for $TESTDIR/.testfile1"
+    test-snapd-sh.with-system-files-plug -c "cat $TESTDIR/testfile1" MATCH "content for $TESTDIR/testfile1"
+    test-snapd-sh.with-system-files-plug -c "ls $TESTDIR/.testdir1"
+    test-snapd-sh.with-system-files-plug -c "ls $TESTDIR/testdir1"
+    test-snapd-sh.with-system-files-plug -c "cat $TESTDIR/.testdir1/.testfilé2" | MATCH "content for $TESTDIR/.testdir1/.testfilé2"
+    test-snapd-sh.with-system-files-plug -c "cat $TESTDIR'/.testdir1/test file2'" | MATCH "content for $TESTDIR/.testdir1/test file2"
+    test-snapd-sh.with-system-files-plug -c "ls $TESTDIR/root/"
+
+    echo "Then the snap is able to write just $TESTDIR/.testdir1 and $TESTDIR/.testfile1"
+    test-snapd-sh.with-system-files-plug -c "echo test >> $TESTDIR/.testfile1"
+    test-snapd-sh.with-system-files-plug -c "touch $TESTDIR/.testdir1/testfilé2"
+
+    if [ "$(snap debug confinement)" = partial ] ; then
+        exit 0
+    fi
+
+    if test-snapd-sh.with-system-files-plug -c "echo test >> $TESTDIR/testfile1" 2> call.error; then
```

Comment:
It made it a little hard to verify that this was correct with it named as 'testfile1'. Naming it 'deniedwrite1' or something might be easier. Not a blocker.